miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/SampleController.java

```diff
@@ -142,6 +151,24 @@ public SampleDto getSample(@PathVariable("id") Long id, UriComponentsBuilder uri
     }
     Long id = null;
     try {
+      if (sampleDto instanceof SampleAliquotDto) {
+        SampleAliquotDto dto = (SampleAliquotDto) sampleDto;
+        if (dto.getParentId() != null) {
+          // Pass
+        } else if (dto.getSampleClassId() == null) {
+          throw new IllegalStateException("No parent and no target sample class.");
```

Comment:
This and the others below should be RestExceptions to form a clear error response with the correct status. This will end up with 500 status (Internal Server Error) when it is really a client-side error if the DTO is missing data